File: 2000.reverse-prefix-of-word.py
#
# @lc app=leetcode id=2000 lang=python3
#
# [2000] Reverse Prefix of Word
#

# @lc code=start
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        i = 0
        n = len(word)
        while i < n and word[i] != ch: i += 1
        return word if i == n else word[i::-1] + word[i + 1:]

# @lc code=end

File: 2000.reverse-prefix-of-word.cs
/*
 * @lc app=leetcode id=2000 lang=csharp
 *
 * [2000] Reverse Prefix of Word
 */

// @lc code=start
public class Solution {
    public string ReversePrefix(string word, char ch) {
        int i = 0;
        int n = word.Length;
        while (i < n && word[i] != ch) {
            i++;
        }
        if (i == n) {
            return word;
        }
        return new string(word.Substring(0, i + 1).Reverse().ToArray()) + word.Substring(i + 1);
    }
}

// @lc code=end

